packages/teams-js/src/public/dialog.ts

```diff
@@ -62,6 +65,34 @@ export namespace dialog {
     sendMessageToParent('tasks.completeTask', [result, Array.isArray(appIds) ? appIds : [appIds]]);
   }
 
+  /**
+   *  The callback to know if the message to parent has been success/failed.
+   *
+   * @param message - The message to send
+   * @param onComplete - The callback to know if the message to parent has been success/failed.
+   */
+  export function sendMessageToParentFromDialog(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    message: any,
+    onComplete?: (status: boolean, reason?: string) => void,
+  ): void {
+    ensureInitialized(FrameContexts.task);
+    sendMessageToParent('messageForParent', [message], onComplete ? onComplete : getGenericOnCompleteHandler());
+  }
+
+  /**
+   * Fucntion to call when an event is received from the Parent
+   *
+   * @param type - The event to listen to. Currently the only supported type is 'message'.
+   * @param listener - listener - The listener that will be called.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export function registerOnMessageFromParent(type: string, listener: (message: any) => void): void {
+    if (type === 'message') {
```

Comment:
Would it make sense to remove the "type" parameter? It doesn't seem to be doing very much for us

Reply:
Yeah, I think we can delete it for this function.
I read it wrong, did all the research and then figured you are asking to change it in our new function, not in the appWindow file. 😂

Reply:
Does it make sense to ensureInitialized here